src/java.base/share/classes/jdk/internal/classfile/constantpool/ClassEntry.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.constantpool;
+
+import java.lang.constant.ClassDesc;
+import java.lang.constant.ConstantDesc;
+import java.util.ArrayList;
+import java.util.List;
+
+import jdk.internal.classfile.impl.AbstractPoolEntry;
+import jdk.internal.classfile.impl.TemporaryConstantPool;
+import jdk.internal.classfile.impl.Util;
+
+
+/**
+ * Models a {@code CONSTANT_Class_info} constant in the constant pool of a
+ * classfile.
+ */
+public sealed interface ClassEntry
+        extends LoadableConstantEntry
+        permits AbstractPoolEntry.ClassEntryImpl {
+
+    @Override
+    default ConstantDesc constantValue() {
+        return asSymbol();
+    }
```

Comment:
It seems possible to use this pattern consistently for `ConstantDynamicEntry`, `MethodHandleEntry`, and `MethodTypeEntry`?

At first i thought why not make `asSymbol` a covariant override of `constantValue`, but its not the same thing, since there are constant values (subtypes of `ConstantValueEntry`) that are not symbols.

Reply:
Yes, I've moved the default `constantValue` delegation to `asSymbol` from implementations up to the  `ConstantDynamicEntry`, `MethodHandleEntry`, and `MethodTypeEntry`.
Thanks.